core/src/main/java/net/sourceforge/jnlp/util/UrlUtils.java

```diff
@@ -341,6 +315,45 @@ public static boolean urlEquals(final URL u1, final URL u2) {
         }
         return false;
     }
+    /**
+     * Checks whether <code>url</code> is relative (or equal) to <code>codebaseUrl</code>.
+     * both urls are processed by sanitizeLastSlash before actual equals. So
+     * protocol://som.url/some/path/ is same as protocol://som.url/some/path. Even
+     * protocol://som.url/some/path\ is same as protocol://som.url/some/path/
+     *
+     * This method returns false in case <code>url</code> contains parent directory notation "..".
+     * See JNLP specification version 9, 3.4: 'A relative URL cannot contain parent directory notations, such as "..". It must denote a file that is stored in a subdirectory of the codebase.'
+     * @param url the url to check
+     * @param codebaseUrl the url to check against
+     * @return true if <code>url</code> is relative to <code>codebaseUrl</code>
+     */
+    public static boolean urlRelativeTo(URL url, URL codebaseUrl) {
+        if (codebaseUrl == url) {
+            return true;
+        }
+        if (codebaseUrl == null || url == null) {
+            return false;
+        }
+        try {
+            URL nu = sanitizeLastSlash(normalizeUrl(url));
```

Comment:
pls make vars final

Reply:
while you are at it, could you rename
- `nu` -> `sanUrl`
- `nup` -> `sanCodebase`

Reply:
Done.